presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/plancheckerprovidermanagers/PlanCheckerProviderManager.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.sanity.plancheckerprovidermanagers;
+
+import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.plan.PlanCheckerProvider;
+import com.facebook.presto.spi.plan.PlanCheckerProviderFactory;
+import com.facebook.presto.spi.plan.SimplePlanFragmentSerde;
+import com.google.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.facebook.presto.util.PropertiesUtil.loadProperties;
+import static java.util.Objects.requireNonNull;
+
+public class PlanCheckerProviderManager
+{
+    private static final File PLAN_CHECKER_PROVIDER_CONFIG = new File("etc/plan-checker-provider.properties");
+    private final NodeManager nodeManager;
+    private final TypeManager typeManager;
+    private final SimplePlanFragmentSerde simplePlanFragmentSerde;
+    private PlanCheckerProviderFactory providerFactory;
+    private PlanCheckerProvider provider;
```

Comment:
You should be able to have multiple PlanCheckerProviders. What if I want the NativePlanChecker and also some other custom checks for my use case?

Also, this needs to be thread safe.  

Look at ConnectorPlanOptimizerManager as an example to mimic.

Reply:
resolved in previous pr